sim/common/wotlk/enchant_effects.go

```diff
@@ -228,6 +242,23 @@ func init() {
 				}
 			},
 		})
+
+		core.RegisterOnItemSwap(3241, func(sim *core.Simulation) {
```

Comment:
Since this exact logic is repeated so much, I think you should be able to add a helper function in `item_swap.go` that reduces this whole block to a 1-line call.